lib/markdown2.py

```diff
@@ -391,8 +391,20 @@ def convert(self, text):
         text += "\n"
 
         rv = UnicodeWithAttrs(text)
-        if "toc" in self.extras:
+        if ("toc" in self.extras or "inline-toc" in self.extras):
+            # Generate TOC HTML as a property to be able to hijack it in "inline-toc" for subsctitution
+            # TODO (Tomas Hubelbauer): See about using that as a static method without a need for a throwaway instance.
             rv._toc = self._toc
+        if "inline-toc" in self.extras:
+            if self._toc[0] is None:
```

Comment:
If we have no headings, there is no need to prepend anything (`toc_html` with be either empty or just `<ul></ul>` I need to check), so this `if` can probably be simplified.

Reply:
I have changed this to only include the TOC HTML if any heading exist.